[18147] xtime will be changed to _timespec64 in an upcoming release of MSVC

### Is there an already existing issue for this?

- [X] I have searched the existing issues

### Expected behavior

No error.

### Current behavior

Hello, I'm a member of Microsoft [VCPKG](https://github.com/microsoft/vcpkg), when I built this port in an internal version of MSVC, it failed with following errors:
```
D:\buildtrees\fastrtps\src\v2.7.0-5b113921c9.clean\include\fastrtps/utils/TimedMutex.hpp(93): error C2065: 'xtime': undeclared identifier
D:\buildtrees\fastrtps\src\v2.7.0-5b113921c9.clean\include\fastrtps/utils/TimedMutex.hpp(93): error C2059: syntax error: ')'
```

### Steps to reproduce

Build with MSVC `19.36.32528.95` and above, `xtime` has been changed to `_timespec64` since https://github.com/microsoft/STL/pull/3594, it's an upcoming release.

### Fast DDS version/commit

`2.7.0`. For details, see: [fastrtps/vcpkg.json](https://github.com/microsoft/vcpkg/blob/master/ports/fastrtps/vcpkg.json)

### Platform/Architecture

Other. Please specify in Additional context section.

### Transport layer

Default configuration, UDPv4 & SHM

### Additional context

* Windows
* Visual Studio

### XML configuration file

_No response_

### Relevant log output

_No response_

### Network traffic capture

_No response_

--- a/include/fastrtps/utils/TimedMutex.hpp
+++ b/include/fastrtps/utils/TimedMutex.hpp
@@ -37,8 +37,13 @@ namespace eprosima {
 namespace fastrtps {
 
 #if defined(_WIN32)
+
 class TimedMutex
 {
+#if defined(_MSC_FULL_VER) && _MSC_FULL_VER >= 193632528
+    using xtime = _timespec64;
+#endif  // _MSC_FULL_VER check
+
 public:
 
     TimedMutex()

--- a/include/fastrtps/utils/TimedMutex.hpp
+++ b/include/fastrtps/utils/TimedMutex.hpp
@@ -40,9 +40,9 @@ namespace fastrtps {
 
 class TimedMutex
 {
-// On MSVC 19.36.32528.95 `xtime` was changed into `_timespec64`.
-// See https://github.com/eProsima/Fast-DDS/issues/3451
-// See https://github.com/microsoft/STL/pull/3594
+    // On MSVC 19.36.32528.95 `xtime` was changed into `_timespec64`.
+    // See https://github.com/eProsima/Fast-DDS/issues/3451
+    // See https://github.com/microsoft/STL/pull/3594
 #if defined(_MSC_FULL_VER) && _MSC_FULL_VER >= 193632528
     using xtime = _timespec64;
 #endif  // _MSC_FULL_VER check

--- a/include/fastrtps/utils/TimedMutex.hpp
+++ b/include/fastrtps/utils/TimedMutex.hpp
@@ -40,6 +40,9 @@ namespace fastrtps {
 
 class TimedMutex
 {
+// On MSVC 19.36.32528.95 `xtime` was changed into `_timespec64`.
+// See https://github.com/eProsima/Fast-DDS/issues/3451
+// See https://github.com/microsoft/STL/pull/3594
 #if defined(_MSC_FULL_VER) && _MSC_FULL_VER >= 193632528
     using xtime = _timespec64;
 #endif  // _MSC_FULL_VER check

